packages/core/src/modules/non-secret/service/NonSecretRecordService.ts

```diff
@@ -0,0 +1,40 @@
+import type { ConnectionRecord } from '../../connections/repository/ConnectionRecord'
+import type { NonSecretRecordTags } from '../repository/NonSecretRecord'
+
+import { Lifecycle, scoped } from 'tsyringe'
+
+import { AriesFrameworkError } from '../../../error'
+import { NonSecretRecord } from '../repository/NonSecretRecord'
+import { NonSecretRecordRepository } from '../repository/NonSecretRecordRepository'
+
+@scoped(Lifecycle.ContainerScoped)
+export class NonSecretRecordService {
+  private nonSecretRepository: NonSecretRecordRepository
+
+  public constructor(nonSecretRepository: NonSecretRecordRepository) {
+    this.nonSecretRepository = nonSecretRepository
+  }
+
+  public async saveRecord(message: string, tags?: NonSecretRecordTags, connectionRecord?: ConnectionRecord) {
```

Comment:
I think for the public api `saveRecord` and the getters should accept a `Record<string, unknown>`. This would allow AFJ to handle the ser/de side and would make it a bit easier for the user as I can imagine that JSON is the preferred storage structure. 

If we want to make the api even nicer, imo, we should add a generic that extends `Record<string, unknown>`. Like so (half pseudo code):

```ts
public async get<R extends Record<string, unknown>(id: string): Promise<NonSecretRecord<R>> {
  const record = /* query */
  record.content = JsonEncoder.fromString(record.content) as R
  return record
}

```

Reply:
I also noticed that a getById, or something similiar is missing. Was this intentional?

Reply:
> I think for the public api `saveRecord` and the getters should accept a `Record<string, unknown>`. This would allow AFJ to handle the ser/de side and would make it a bit easier for the user as I can imagine that JSON is the preferred storage structure.
> 
> If we want to make the api even nicer, imo, we should add a generic that extends `Record<string, unknown>`. Like so (half pseudo code):
> 
> ```ts
> public async get<R extends Record<string, unknown>(id: string): Promise<NonSecretRecord<R>> {
>   const record = /* query */
>   record.content = JsonEncoder.fromString(record.content) as R
>   return record
> }
> ```

That would be very useful to make this module even more generic!

Reply:
Good point, can we do that enhancement as next level? I not quit familiar with that and it seems causing lots of errors when adding it

Reply:
I can't find any similar implementation in any of the current modules using this Generic way. 
@blu3beri  Can you suggest please the change to make to "GenericRecord" so that it can be wrapped in the function you suggested?